build-support/bin/xz/linux/x86_64/5.2.4/build-xz.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```

Comment:
This was updated: see the README.md

Reply:
So the choice was made in #59 to do this instead of the previous workflow -- one benefit is that scripts don't have to know where their output is supposed to go, but can just generate everything in the current directory (a leaf subdirectory of `build-support/bin/`). Another benefit is that we have a specific script explicitly bound to a specific version of a specific package, without polluting the root dir with explicitly-versioned files (which can lead to mistakes when trying to edit a particular one). I can't remember where this happened in, but I'm not 100% about having the same script copied to both linux and osx dirs, especially since here and in previous PRs I have made sure to make a script which accepts the platform and the version as command-line arguments (e.g. `build-gcc.sh`), and then just invoked that script for the specific platform and version in a `build.sh` script, as done here. This is something I will continue to do (which is, as you described, one unified build with parameters, if the `build-gcc.sh` script is kept in sync across different versions and platforms).

I'm mostly fine with this because it does make "reproducibility" easier without introducing a ton of overhead or processes. I think the eventual solution should probably be to introduce some CI on this repo (or something similar) which tests it against multiple versions of pants -- we don't have any guarantees that pants-provided binaries downloaded today will work with future versions of pants which may expect a different binary for the same requested version (this would happen for example if we removed some files from a provided binary for whatever reason). Until then, I'm mostly fine with the current (new) approach (with scripts colocated with the binaries).

I think we should at some point define what level of compatibility pants-provided binaries are supposed to have with different versions of pants, and then canonicalize that with a small test for some set of supported pants versions, for each version of each package provided. We would likely prune old or unused binaries that are still around as we performed that process. I don't know how to do this effectively, or if the actual checks we perform for each binary are something we would want to put into pants proper, so users who download from their own set of `--binaries-baseurls`, say on an internal network, will be able to perform the same checks.

Reply:
ah, k. thanks for the explanation.